lib/incoming_mail/handler.rb

```diff
@@ -0,0 +1,98 @@
+module IncomingMail
+  class Handler
+
+    def initialize(params = {})
+      params.each do |key, value|
+        instance_variable_set("@#{key}", value)
+      end
+    end
+
+    def handle(payload)
+      # raw JSON
+      if payload.is_a?(Array) and payload[0]['event'] and payload[0]['event'] == 'inbound'
```

Comment:
this is super small / not a blocker but one idea in a situation like this would be to have a private method called `is_raw_json?` and include `payload.is_a?(Array) and payload[0]['event'] and payload[0]['event'] == 'inbound'` in there so the code itself explains the branching here (same for the `elsif`)